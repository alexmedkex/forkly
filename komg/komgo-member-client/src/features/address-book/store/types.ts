import { List } from 'immutable'
import { ImmutableMap } from '../../../utils/types'
import { Action } from 'redux'
import { IMember } from '../../members/store/types'

export enum AddressBookActionType {
  GET_COMPANIES_REQUEST = '@@addressBook/GET_COMPANIESS_REQUEST',
  GET_COMPANIES_SUCCESS = '@@addressBook/GET_COMPANIESS_SUCCESS',
  GET_COMPANIES_FAILURE = '@@addressBook/GET_COMPANIESS_FAILURE',
  GET_COMPANY_REQUEST = '@@addressBook/GET_COMPANY_REQUEST',
  GET_COMPANY_SUCCESS = '@@addressBook/GET_COMPANY_SUCCESS',
  GET_COMPANY_FAILURE = '@@addressBook/GET_COMPANY_FAILURE',
  GENERATE_MEMBER_PACKAGE_REQUEST = '@@addressBook/GENERATE_MEMBER_PACKAGE_REQUEST',
  GENERATE_MEMBER_PACKAGE_SUCCESS = '@@addressBook/GENERATE_MEMBER_PACKAGE_SUCCESS',
  GENERATE_MEMBER_PACKAGE_FAILURE = '@@addressBook/GENERATE_MEMBER_PACKAGE_FAILURE',
  ADD_COMPANY_TO_ENS_REQUEST = '@@addressBook/ADD_COMPANY_TO_ENS_REQUEST',
  ADD_COMPANY_TO_ENS_SUCCESS = '@@addressBook/ADD_COMPANY_TO_ENS_SUCCESS',
  ADD_COMPANY_TO_ENS_FAILURE = '@@addressBook/ADD_COMPANY_TO_ENS_FAILURE',
  CONFIGURE_MQ_REQUEST = '@@addressBook/CONFIGURE_MQ_REQUEST',
  CONFIGURE_MQ_SUCCESS = '@@addressBook/CONFIGURE_MQ_SUCCESS',
  CONFIGURE_MQ_FAILURE = '@@addressBook/CONFIGURE_MQ_FAILURE',
  CREATE_COMPANY_REQUEST = '@@addressBook/CREATE_COMPANY_REQUEST',
  CREATE_COMPANY_SUCCESS = '@@addressBook/CREATE_COMPANY_SUCCESS',
  CREATE_COMPANY_FAILURE = '@@addressBook/CREATE_COMPANY_FAILURE',
  UPDATE_COMPANY_REQUEST = '@@addressBook/UPDATE_COMPANY_REQUEST',
  UPDATE_COMPANY_SUCCESS = '@@addressBook/UPDATE_COMPANY_SUCCESS',
  UPDATE_COMPANY_FAILURE = '@@addressBook/UPDATE_COMPANY_FAILURE',
  DeleteCompanyByIdSuccess = '@@addressBook/DELETE_COMPANY_BY_ID_SUCCESS'
}

export interface AddressBookStateProperties {
  companies: List<IMember>
}

export type AddressBookState = ImmutableMap<AddressBookStateProperties>

export interface GetCompaniesRequest extends Action {
  type: AddressBookActionType.GET_COMPANIES_SUCCESS
  payload: IMember[]
}
export interface GetCompanyRequest extends Action {
  type: AddressBookActionType.GET_COMPANY_SUCCESS
  payload: IMember
}
export interface GenerateMemberPackageRequest extends Action {
  type: AddressBookActionType.GENERATE_MEMBER_PACKAGE_SUCCESS
  payload: IMember
}
export interface AddCompanyToENSRequest extends Action {
  type: AddressBookActionType.ADD_COMPANY_TO_ENS_SUCCESS
  payload: IMember
}
export interface ConfigureMQRequest extends Action {
  type: AddressBookActionType.CONFIGURE_MQ_SUCCESS
  payload: IMember
}

export interface CreateCompanyRequest extends Action {
  type: AddressBookActionType.CREATE_COMPANY_SUCCESS
  payload: IMember
}
export interface UpdateCompanyRequest extends Action {
  type: AddressBookActionType.UPDATE_COMPANY_SUCCESS
  payload: IMember
}

export interface DeleteCompanyByIdSuccess extends Action {
  type: AddressBookActionType.DeleteCompanyByIdSuccess
  payload: { id: string }
}

export type AddressBookActions =
  | GetCompaniesRequest
  | GetCompanyRequest
  | CreateCompanyRequest
  | UpdateCompanyRequest
  | GenerateMemberPackageRequest
  | AddCompanyToENSRequest
  | ConfigureMQRequest
  | DeleteCompanyByIdSuccess
