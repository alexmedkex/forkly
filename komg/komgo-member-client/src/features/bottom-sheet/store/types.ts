import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import {
  CreateDocumentError,
  CreateDocumentSuccess,
  DocumentRegisteredError,
  DocumentRegisteredSuccess,
  HasId,
  HasName
} from '../../document-management/store/types'
import { ActionCreator } from 'react-redux'

export enum BottomSheetActionType {
  REMOVE_BOTTOMSHEET_ITEM = '@@btsh/REMOVE_BOTTOMSHEET_ITEM',
  RETRY_BOTTOMSHEET_ITEM = '@@btsh/RETRY_BOTTOMSHEET_ITEM',
  UPDATE_BOTTOMSHEET_ITEM = '@@btsh/UPDATE_BOTTOMSHEET_ITEM',
  CREATE_BOTTOMSHEET_ITEM = '@@btsh/CREATE_BOTTOMSHEET_ITEM'
}

export interface BottomSheetItem extends HasId, HasName {
  state: BottomSheetStatus
  itemType?: BottomSheetItemType
  displayStatus?: string
  action?: ActionCreator<any>
  navigationLink?: string
}

export enum BottomSheetItemType {
  REGISTER_KYC_DOCUMENT,
  CREATE_MAGIC_LINK,
  REGISTER_RD_DOCUMENT,
  REGISTER_TRADE_DOCUMENT
}

export enum BottomSheetStatus {
  NEW = 'NEW',
  PENDING = 'PENDING',
  REGISTERED = 'REGISTERED',
  FAILED = 'FAILED'
}

export interface BottomSheetStateFields {
  visible: boolean
  items: BottomSheetItem[]
}

export interface RemoveBottomSheetItem extends Action {
  type: BottomSheetActionType.REMOVE_BOTTOMSHEET_ITEM
  payload: HasId
}

export interface RetryRegisterItem extends Action {
  type: BottomSheetActionType.RETRY_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface UpdateBottomSheetItem extends Action {
  type: BottomSheetActionType.UPDATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface CreateBottomSheetItem extends Action {
  type: BottomSheetActionType.CREATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export type BottomSheetState = ImmutableMap<BottomSheetStateFields>

export type BottomSheetAction =
  | DocumentRegisteredSuccess
  | DocumentRegisteredError
  | CreateDocumentSuccess
  | CreateDocumentError
  | RemoveBottomSheetItem
  | RetryRegisterItem
  | UpdateBottomSheetItem
  | CreateBottomSheetItem
