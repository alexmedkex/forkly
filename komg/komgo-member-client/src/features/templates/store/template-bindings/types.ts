import { ITemplateBinding } from '@komgo/types'
import { Map, List } from 'immutable'
import { ImmutableMap } from '../../../../utils/types'
import { Action } from 'redux'

export enum EditorTemplateBindingsActionType {
  GET_TEMPLATE_BINDING_REQUEST = '@@template-bindings/GET_TEMPLATE_BINDING_REQUEST',
  GET_TEMPLATE_BINDING_SUCCESS = '@@template-bindings/GET_TEMPLATE_BINDING_SUCCESS',
  GET_TEMPLATE_BINDING_FAILURE = '@@template-bindings/GET_TEMPLATE_BINDING_FAILURE',
  FETCH_TEMPLATE_BINDINGS_REQUEST = '@@template-bindings/FETCH_TEMPLATE_BINDINGS_REQUEST',
  FETCH_TEMPLATE_BINDINGS_SUCCESS = '@@template-bindings/FETCH_TEMPLATE_BINDINGS_SUCCESS',
  FETCH_TEMPLATE_BINDINGS_FAILURE = '@@template-bindings/FETCH_TEMPLATE_BINDINGS_FAILURE'
}

export interface GetTemplateBindingsSuccessAction extends Action {
  type: EditorTemplateBindingsActionType.GET_TEMPLATE_BINDING_SUCCESS
  payload: ITemplateBinding
}

export interface FetchTemplateBindingsSuccessAction extends Action {
  type: EditorTemplateBindingsActionType.FETCH_TEMPLATE_BINDINGS_SUCCESS
  payload: { limit: number; skip: number; total: number; items: ITemplateBinding[] }
}

export interface TemplateBindingsStateProperties {
  byStaticId: Map<string, Map<keyof ITemplateBinding, any>>
  staticIds: List<string>
  total: number
}

export type EditorTemplateBindingsState = ImmutableMap<TemplateBindingsStateProperties>
