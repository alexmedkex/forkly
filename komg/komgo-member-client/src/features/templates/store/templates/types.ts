import { ITemplate } from '@komgo/types'
import { Map, List } from 'immutable'
import { ImmutableMap } from '../../../../utils/types'
import { Action } from 'redux'

export enum EditorTemplatesActionType {
  FETCH_TEMPLATES_REQUEST = '@@templates/FETCH_TEMPLATES_REQUEST',
  FETCH_TEMPLATES_SUCCESS = '@@templates/FETCH_TEMPLATES_SUCCESS',
  FETCH_TEMPLATES_FAILURE = '@@templates/FETCH_TEMPLATES_FAILURE',
  CREATE_TEMPLATE_REQUEST = '@@templates/CREATE_TEMPLATE_REQUEST',
  CREATE_TEMPLATE_SUCCESS = '@@templates/CREATE_TEMPLATE_SUCCESS',
  CREATE_TEMPLATE_FAILURE = '@@templates/CREATE_TEMPLATE_FAILURE',
  UPDATE_TEMPLATE_REQUEST = '@@templates/UPDATE_TEMPLATE_REQUEST',
  UPDATE_TEMPLATE_SUCCESS = '@@templates/UPDATE_TEMPLATE_SUCCESS',
  UPDATE_TEMPLATE_FAILURE = '@@templates/UPDATE_TEMPLATE_FAILURE',
  GET_TEMPLATE_REQUEST = '@@templates/GET_TEMPLATE_REQUEST',
  GET_TEMPLATE_SUCCESS = '@@templates/GET_TEMPLATE_SUCCESS',
  GET_TEMPLATE_FAILURE = '@@templates/GET_TEMPLATE_FAILURE',
  DELETE_TEMPLATE_REQUEST = '@@templates/DELETE_TEMPLATE_REQUEST',
  DELETE_TEMPLATE_SUCCESS = '@@templates/DELETE_TEMPLATE_SUCCESS',
  DELETE_TEMPLATE_FAILURE = '@@templates/DELETE_TEMPLATE_FAILURE'
}

export interface FetchTemplatesSuccessAction extends Action {
  type: EditorTemplatesActionType.FETCH_TEMPLATES_SUCCESS
  payload: { limit: number; skip: number; total: number; items: ITemplate[] }
}

export interface TemplateStateProperties {
  byStaticId: Map<string, Map<keyof ITemplate, any>>
  staticIds: List<string>
  total: number
}

export type EditorTemplatesState = ImmutableMap<TemplateStateProperties>
