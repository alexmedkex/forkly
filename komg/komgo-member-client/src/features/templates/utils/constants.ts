export const EMPTY_TEMPLATE = {
  object: 'value',
  document: {
    object: 'document',
    data: {},
    nodes: [
      {
        object: 'block',
        type: 'paragraph',
        data: {},
        nodes: [
          {
            object: 'text',
            text: 'Empty Template',
            marks: [{ object: 'mark', type: 'bold', data: {} }]
          }
        ]
      },
      {
        object: 'block',
        type: 'paragraph',
        data: {},
        nodes: [
          {
            object: 'text',
            text: '',
            marks: [{ object: 'mark', type: 'bold', data: {} }]
          }
        ]
      },
      {
        object: 'block',
        type: 'paragraph',
        data: {},
        nodes: [
          {
            object: 'text',
            text: 'Please add your wording then add the dynamic fields you need from the above',
            marks: []
          },
          {
            object: 'text',
            text: ' Add Field',
            marks: [{ object: 'mark', type: 'bold', data: {} }]
          },
          { object: 'text', text: ' button', marks: [] }
        ]
      },
      {
        object: 'block',
        type: 'paragraph',
        data: {},
        nodes: [{ object: 'text', text: '', marks: [] }]
      },
      {
        object: 'block',
        type: 'paragraph',
        data: {},
        nodes: [
          {
            object: 'text',
            text: 'e.g. Some text then a variable: ',
            marks: []
          },
          {
            object: 'inline',
            type: 'variable',
            data: { path: 'applicant.x500Name.CN', title: 'Applicant name' },
            nodes: [{ object: 'text', text: 'Applicant name', marks: [] }]
          },
          { object: 'text', text: '', marks: [] }
        ]
      }
    ]
  }
}
