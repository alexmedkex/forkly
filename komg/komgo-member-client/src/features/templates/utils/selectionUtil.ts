import { parse } from 'qs'

export interface ISelection {
  select: boolean
  redirectTo: string
  type: string // TODO LS we need to decide how to mark the template. I don't think LetterOfCreditType is the right one
}

const REDIRECT_WHITE_LIST = [/^\/letters-of-credit\/new/]

export const buildSelection = (querystring: string): ISelection => {
  const { select = false, type, redirectTo } = parse(querystring, {
    ignoreQueryPrefix: true,
    decoder: str => {
      const keywords = {
        true: true,
        false: false,
        null: null,
        undefined
      }
      if (str in keywords) {
        return keywords[str]
      }
      return decodeURIComponent(str)
    }
  })

  if (redirectTo && !REDIRECT_WHITE_LIST.some(route => redirectTo.match(route))) {
    throw new Error(`Redirect to ${redirectTo} not supported`)
  }

  return {
    redirectTo,
    select,
    type
  }
}
