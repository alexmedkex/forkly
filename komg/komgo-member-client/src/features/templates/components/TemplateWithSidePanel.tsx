import React from 'react'
import { ViewContainer } from './ViewContainer'
import { SidePanel, SidePanelInner } from './SidePanel'
import { Preview } from './Preview'
import { Template, ITemplateOptions } from '@komgo/ui-components'
import { ReactNode } from 'react-redux'
import DocumentContentView from '../../document-management/components/documents/DocumentContentView'
import { ImmutableMap } from '../../../utils/types'
import { Document, DocumentStateFields } from '../../document-management'

export interface TemplateWithSidePanelProps extends ITemplateOptions {
  children?: ReactNode
  issuanceDocument?: ImmutableMap<DocumentStateFields>
  issuanceDocumentMetadata?: Document
}

export const TemplateWithSidePanel = ({
  children,
  issuanceDocument,
  issuanceDocumentMetadata,
  ...rest
}: TemplateWithSidePanelProps) => {
  return (
    <ViewContainer>
      {children && (
        <SidePanel>
          <SidePanelInner>{children}</SidePanelInner>
        </SidePanel>
      )}

      <Preview height={'80px'}>
        {issuanceDocumentMetadata ? (
          <DocumentContentView
            documentContent={issuanceDocument.get('documentRaw')}
            isLoadingContent={issuanceDocument.get('isLoadingContent')}
            documentType={issuanceDocument.get('documentType')}
          />
        ) : (
          <Template {...rest} />
        )}
      </Preview>
    </ViewContainer>
  )
}
