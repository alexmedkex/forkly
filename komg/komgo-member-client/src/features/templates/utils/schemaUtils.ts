import { findLetterOfCreditSchema } from '../../letter-of-credit/utils/findLetterOfCreditSchema'

export const resolveBindings = (
  bindings: { [key: string]: string },
  schemaResolver: (key: string) => any = findLetterOfCreditSchema
): { [key: string]: any } =>
  Object.entries(bindings).reduce(
    (memo, [key, value]) => ({
      ...memo,
      [key]: schemaResolver(value)
    }),
    {}
  )

export const findTemplateBindingType = (schemaId: string) => {
  // FIXME LS until we don't add a templateBiding name
  return schemaId.match(/http:\/\/komgo.io\/schema\/(.*)\/template-bindings/)[1].toUpperCase()
}
