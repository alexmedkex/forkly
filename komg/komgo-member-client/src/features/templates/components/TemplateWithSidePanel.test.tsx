import * as React from 'react'
import { buildFields } from '@komgo/types'
import { render } from '@testing-library/react'
import { TemplateWithSidePanel } from './TemplateWithSidePanel'
import { buildFakeTemplateInstance } from '@komgo/types/dist/template-library/template/faker'
import { resolveBindings } from '../utils/schemaUtils'
import { findLetterOfCreditSchema } from '../../letter-of-credit/utils/findLetterOfCreditSchema'
import { FILTERED_FIELDS } from '../../letter-of-credit/constants'
import { Product } from '../../document-management'
import { fromJS } from 'immutable'

const templateInstance = buildFakeTemplateInstance()

const fields = buildFields({
  bindings: resolveBindings(templateInstance.bindings, findLetterOfCreditSchema),
  dataSchema: findLetterOfCreditSchema(templateInstance.dataSchemaId),
  templateSchema: findLetterOfCreditSchema(templateInstance.templateSchemaId)
}).filter(f => !FILTERED_FIELDS.includes(f.dataPath))

const globalAsAny = global as any
const originalGetSelection = globalAsAny.window.getSelection

describe('TemplateWithSidePanel', () => {
  beforeEach(() => {
    globalAsAny.window.getSelection = jest.fn(() => {
      return {}
    })
  })
  afterEach(() => {
    globalAsAny.window.getSelection = originalGetSelection
  })
  it('matches snapshot with a template and a side panel', () => {
    const sidePanel = <div>my side panel content</div>

    const { asFragment } = render(
      <TemplateWithSidePanel template={templateInstance.template} data={templateInstance.data} fields={fields}>
        {sidePanel}
      </TemplateWithSidePanel>
    )

    expect(asFragment()).toMatchSnapshot()
  })
  it('matches snapshot without a child element', () => {
    const { asFragment } = render(
      <TemplateWithSidePanel template={templateInstance.template} data={templateInstance.data} fields={fields} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
  it('matches snapshot with a document', () => {
    const product: Product = {
      id: 'tradeFinance',
      name: 'productName'
    }

    const issuanceDocumentMetadata = true as any

    const issuanceDocument = fromJS({
      documentRaw: 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mOs1Vv6HwAE0QJRJ/jTdwAAAABJRU5ErkJggg==',
      isLoadingContent: false,
      documentType: 'image/png'
    })

    const { asFragment } = render(
      <TemplateWithSidePanel
        template={templateInstance.template}
        data={templateInstance.data}
        fields={fields}
        issuanceDocumentMetadata={issuanceDocumentMetadata}
        issuanceDocument={issuanceDocument}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
