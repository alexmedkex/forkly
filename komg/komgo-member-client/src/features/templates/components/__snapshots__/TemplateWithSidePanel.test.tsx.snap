// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateWithSidePanel matches snapshot with a document 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
  width: 100%;
  border-top: 1px solid #c0cfde;
  background-color: #f2f5f8;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  background-color: #f2f5f8;
}

@media (min-width:68.75em) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    height: 100%;
    overflow: none;
  }
}

@media (min-width:68.75em) {
  .c1 {
    max-width: 990px;
    margin-left: auto;
    margin-right: auto;
    padding: 16px;
    height: calc(100vh - 80px);
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
      height="80px"
    >
      <img
        class="ui image"
        src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mOs1Vv6HwAE0QJRJ/jTdwAAAABJRU5ErkJggg=="
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TemplateWithSidePanel matches snapshot with a template and a side panel 1`] = `
<DocumentFragment>
  .c8 {
  background: rgba(148,109,192,0.4);
  font-style: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: normal;
  text-transform: uppercase;
}

.c5 {
  width: 100%;
  padding: 8px;
  min-height: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0 4px 2px -2px rgba(192,207,222,0.51);
}

.c6 {
  margin-left: auto;
}

.c7 {
  margin-top: 2px;
  padding: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #c0cfde;
  background-color: white;
  box-shadow: 0 4px 2px -2px rgba(192,207,222,0.51);
  height: 100%;
  width: 100%;
}

.c0 {
  margin-top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
  width: 100%;
  border-top: 1px solid #c0cfde;
  background-color: #f2f5f8;
}

.c1 {
  width: 100%;
  background-color: #f2f5f8;
  box-shadow: 0 4px 2px -2px rgba(192,207,222,0.51);
}

.c2 {
  padding: 16px;
  overflow: auto;
  height: 100%;
  background-color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #c0cfde;
  box-shadow: 0 4px 2px -2px rgba(192,207,222,0.51);
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  background-color: #f2f5f8;
}

@media (min-width:68.75em) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    height: 100%;
    overflow: none;
  }
}

@media (min-width:68.75em) {
  .c1 {
    padding: 16px 0 16px 16px;
    -webkit-scroll-behavior: smooth;
    -moz-scroll-behavior: smooth;
    -ms-scroll-behavior: smooth;
    scroll-behavior: smooth;
    height: calc(100vh - 80px);
    margin: 0;
    box-shadow: none;
    width: 400px;
    max-width: 400px;
    min-width: 400px;
  }
}

@media (min-width:68.75em) {
  .c3 {
    max-width: 990px;
    margin-left: auto;
    margin-right: auto;
    padding: 16px;
    height: calc(100vh - 80px);
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div>
          my side panel content
        </div>
      </div>
    </div>
    <div
      class="c3"
      height="80px"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            style="padding-left: 16px; color: rgb(212, 92, 100);"
          >
            <i
              aria-hidden="true"
              class="red warning sign icon"
            />
            <span
              style="padding-left: 16px;"
            >
              The template is missing 8 mandatory fields. 
              <span
                data-test-id="resolve-missing-fields"
                style="color: rgb(212, 92, 100); text-decoration: underline; cursor: pointer;"
              >
                Resolve
              </span>
            </span>
          </div>
          <div
            class="c6"
          >
            <div
              style="display: flex; align-items: center;"
            >
              <div
                class="ui fitted toggle checkbox"
                data-test-id="button-mode"
              >
                <input
                  class="hidden"
                  readonly=""
                  tabindex="0"
                  type="checkbox"
                  value=""
                />
                <label />
              </div>
              <label
                style="padding-right: 16px; padding-left: 16px;"
              >
                Edit mode
              </label>
            </div>
          </div>
        </div>
        <div
          class="c7"
        >
          <div
            autocorrect="on"
            data-gramm="false"
            data-key="0"
            data-slate-editor="true"
            spellcheck="true"
            style="outline: none; white-space: pre-wrap; word-wrap: break-word; width: 100%; height: 100%; flex-grow: 1;"
          >
            <div
              data-key="1"
              data-slate-object="block"
              style="position: relative;"
            >
              <span
                data-key="2"
                data-slate-object="text"
              >
                <span
                  data-offset-key="2:0"
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    some text 
                  </span>
                </span>
              </span>
              <span
                data-key="3"
                data-slate-void="true"
              >
                <span>
                  <var
                    class="c8"
                    data-key="3"
                    data-slate-object="inline"
                    data-test-id="var-trade.quantity"
                  >
                    600000
                  </var>
                </span>
              </span>
              <span
                data-key="5"
                data-slate-object="text"
              >
                <span
                  data-offset-key="5:0"
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                     
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TemplateWithSidePanel matches snapshot without a child element 1`] = `
<DocumentFragment>
  .c6 {
  background: rgba(148,109,192,0.4);
  font-style: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: normal;
  text-transform: uppercase;
}

.c3 {
  width: 100%;
  padding: 8px;
  min-height: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0 4px 2px -2px rgba(192,207,222,0.51);
}

.c4 {
  margin-left: auto;
}

.c5 {
  margin-top: 2px;
  padding: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #c0cfde;
  background-color: white;
  box-shadow: 0 4px 2px -2px rgba(192,207,222,0.51);
  height: 100%;
  width: 100%;
}

.c0 {
  margin-top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
  width: 100%;
  border-top: 1px solid #c0cfde;
  background-color: #f2f5f8;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  background-color: #f2f5f8;
}

@media (min-width:68.75em) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    height: 100%;
    overflow: none;
  }
}

@media (min-width:68.75em) {
  .c1 {
    max-width: 990px;
    margin-left: auto;
    margin-right: auto;
    padding: 16px;
    height: calc(100vh - 80px);
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
      height="80px"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            style="padding-left: 16px; color: rgb(212, 92, 100);"
          >
            <i
              aria-hidden="true"
              class="red warning sign icon"
            />
            <span
              style="padding-left: 16px;"
            >
              The template is missing 8 mandatory fields. 
              <span
                data-test-id="resolve-missing-fields"
                style="color: rgb(212, 92, 100); text-decoration: underline; cursor: pointer;"
              >
                Resolve
              </span>
            </span>
          </div>
          <div
            class="c4"
          >
            <div
              style="display: flex; align-items: center;"
            >
              <div
                class="ui fitted toggle checkbox"
                data-test-id="button-mode"
              >
                <input
                  class="hidden"
                  readonly=""
                  tabindex="0"
                  type="checkbox"
                  value=""
                />
                <label />
              </div>
              <label
                style="padding-right: 16px; padding-left: 16px;"
              >
                Edit mode
              </label>
            </div>
          </div>
        </div>
        <div
          class="c5"
        >
          <div
            autocorrect="on"
            data-gramm="false"
            data-key="14"
            data-slate-editor="true"
            spellcheck="true"
            style="outline: none; white-space: pre-wrap; word-wrap: break-word; width: 100%; height: 100%; flex-grow: 1;"
          >
            <div
              data-key="15"
              data-slate-object="block"
              style="position: relative;"
            >
              <span
                data-key="16"
                data-slate-object="text"
              >
                <span
                  data-offset-key="16:0"
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    some text 
                  </span>
                </span>
              </span>
              <span
                data-key="17"
                data-slate-void="true"
              >
                <span>
                  <var
                    class="c6"
                    data-key="17"
                    data-slate-object="inline"
                    data-test-id="var-trade.quantity"
                  >
                    600000
                  </var>
                </span>
              </span>
              <span
                data-key="19"
                data-slate-object="text"
              >
                <span
                  data-offset-key="19:0"
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                     
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
